File: 96/96_version1.py
class Solution:
    def numTrees(self, n: int) -> int:
        result = [0]*(n+1)
        result[0] = 1
        result[1] = 1
        for N in range(2,n+1):
            for i in range(1,N+1):
                result[N]+=result[i-1]*result[N-i]
        return result[n]

File: 96/96_version1.java
class Solution {
    public int numTrees(int n) {
        int[] G = new int[n+1];
        G[0] = 1;G[1] = 1;
        for(int N = 2;N<=n;N++){
            for(int i=1;i<=N;i++){
                G[N]+=G[i-1]*G[N-i];
            }
        }
        return G[n];
    }
}
